models/sipnet/R/write.configs.SIPNET.R

```diff
@@ -490,20 +489,41 @@ write.config.SIPNET <- function(defaults, trait.values, settings, run.id, inputs
     }
   } ## end loop over PFTS
   ####### end parameter update
-  #working on reading soil file (only working for 1 soil file)
-  if(length(settings$run$inputs$soilinitcond$path)==1){
-    soil_IC_list <- PEcAn.data.land::pool_ic_netcdf2list(settings$run$inputs$soilinitcond$path)
-    #SoilWHC and LitterWHC
-    if("volume_fraction_of_water_in_soil_at_saturation"%in%names(soil_IC_list$vals)){
-      #SoilWHC
-      param[which(param[, 1] == "soilWHC"), 2] <- mean(unlist(soil_IC_list$vals["volume_fraction_of_water_in_soil_at_saturation"]))*100
-      
-      #LitterWHC
-      #param[which(param[, 1] == "litterWHC"), 2] <- unlist(soil_IC_list$vals["volume_fraction_of_water_in_soil_at_saturation"])[1]*100
+  #working on reading soil file
+  if (length(settings$run$inputs$soil_physics$path) > 0) {
+    template.soil_physics <- settings$run$inputs$soil_physics$path  ## read from settings
+    
+    if (!is.null(inputs)) {
+      ## override if specified in inputs
+      if ("soil_physics" %in% names(inputs)) {
+        template.soil_physics <- inputs$soil_physics$path
+      }
     }
-    if("soil_hydraulic_conductivity_at_saturation"%in%names(soil_IC_list$vals)){
-      #litwaterDrainrate
-      param[which(param[, 1] == "litWaterDrainRate"), 2] <- unlist(soil_IC_list$vals["soil_hydraulic_conductivity_at_saturation"])[1]*100/(3600*24)
+    
+    if (length(template.soil_physics)!=1) {
+      PEcAn.logger::logger.warn(
+        paste0("No single soil physical parameter file was found for ",
+               run.id))
+    } else {
+    soil_IC_list <- PEcAn.data.land::pool_ic_netcdf2list(template.soil_physics)
+    # Calculate the thickness of soil layers based on the assumption that the depth values are at bottoms and the first layer top is at 0
+    if ("depth" %in% names(soil_IC_list$dims)) {
+      thickness<-c(soil_IC_list$dims$depth[1],diff(soil_IC_list$dims$depth))
+      #SoilWHC
+      if ("volume_fraction_of_water_in_soil_at_saturation" %in% names(soil_IC_list$vals)) {
+         # Calculate the soilWHC for the whole soil profile in cm
+         soilWHC_total <- sum(unlist(soil_IC_list$vals["volume_fraction_of_water_in_soil_at_saturation"])*thickness)
+         param[which(param[, 1] == "soilWHC"), 2] <- soilWHC_total
+         #LitterWHC in cm (top soil layer)
+         param[which(param[, 1] == "litterWHC"), 2] <- unlist(soil_IC_list$vals["volume_fraction_of_water_in_soil_at_saturation"])[1]*thickness[1]
```

Comment:
We should be very careful here  around assumptions about the inputs -- depending on the soil product there's no guarantee that the first thickness will be the litter layer. Indeed, I could easily see cases where the first layer is 0-30 cm or where there's only 1 layer provided (e.g. first and only thickness is 0-100). Also, your current code seems to double count the first layer in both the soilWHC and litterWHC.